include/sys/boardctl.h

```diff
@@ -253,6 +261,26 @@ struct boardioc_pm_ctrl_s
 };
 #endif
 
+#ifdef CONFIG_BOARDCTL_NETCONF
+/* Describes the network configuration */
+
+enum boardioc_netconf_e
+{
+  BOARDIOC_NETCONF_DHCP     = 0x1, /* Use DHCP */
+  BOARDIOC_NETCONF_STATIC   = 0x2, /* Use static IP */
+  BOARDIOC_NETCONF_FALLBACK = 0x3, /* Use DHCP with fall back static IP */
+};
+
+struct boardioc_netconf_s
+{
+  enum boardioc_netconf_e flags; /* Configure for static and/or DHCP */
```

Comment:
Why need return DHCP/STATIC/FALLBACK from board? it's enough to add FALLBACK Kconfig. Board just need return the saved address.

Reply:
So it can be set at run time and changed at run time.

Reply:
> 
> 
> So it can be set at run time and changed at run time.

All of the OS settings in boardioc_netconf_s can be managed with existing ioctl calls (or using the helpers in netlib).  Those are totally redundant.

And the flags field.  I object strenuously in the that.  That MUST not be used.   There must NOT be any OS interface of DHCP.  DHCP is purely an application, user-space thing and must never contaminate the OS/application interface.  This can never come into the OS.